### PR TITLE
Upgrade to Laravel 5.6

### PR DESCRIPTION
**Pull request by DarkaOnLine**
- Updated composer dependencies and config files.
- Fixed bug - when Best Reply is deleted its `id` is not unset from thread `best_reply_id`.
- Fix test to reflect Laravel 5.6 changes. 201 status code is returned for newly created models instead of 200

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,24 +1,24 @@
 {
     "name": "laravel/laravel",
```
Comment (m1guelpf) [inline composer.json]: Maybe change this to `laracasts/council`?

File: app/Thread.php [hunk 2]
```diff
@@ -261,6 +261,17 @@ public function markBestReply(Reply $reply)
         $reply->owner->gainReputation('best_reply_awarded');
     }
 
+    /**
+     * Unset best reply from thread.
+     */
+    public function unsetBestReplay()
```
Comment (goncalomcunha) [inline app/Thread.php]: The name of the function has a typo

File: composer.json (hunk 1 above)
Comment (DarkaOnLine) [reply composer.json]: Thought about that. Will leave for Jeffrey to decide which to use: 'laracasts/council' or `JeffreyWay/council` or  'council/council' :)

File: app/Thread.php (hunk 2 above)
Comment (DarkaOnLine) [reply app/Thread.php]: Fixed

File: app/Reply.php [hunk 3]
```diff
@@ -43,6 +43,12 @@ protected static function boot()
             $reply->owner->gainReputation('reply_posted');
         });
 
+        static::deleting(function ($reply) {
+            if ($reply->isBest()) {
+                $reply->thread->unsetBestReplay();
+            }
+        });
+
```
Comment (DarkaOnLine) [inline app/Reply.php]: Can someone explain this for me :) In the migrations we have foreign key to set best_reply_id to null on reply delete: https://github.com/JeffreyWay/council/blob/master/database/migrations/2017_03_15_175959_create_threads_table.php#L30-L34

On Laravel 5.5 [this test works](https://github.com/JeffreyWay/council/blob/master/tests/Feature/BestReplyTest.php#L47-L58), but on 5.6 it doesn't :( Why?

File: tests/Feature/PinThreadsTest.php [hunk 4]
```diff
@@ -56,22 +56,16 @@ public function pinned_threads_are_listed_first()
...
-                ['id' => $ids[1]],
-                ['id' => $ids[2]]
-            ]
-        ]);
-        
+        $response_data = $this->getJson(route('threads'))->decodeResponseJson()['data'];
+        $this->assertEquals($ids[0], $response_data[0]['id']);
+        $this->assertEquals($ids[1], $response_data[1]['id']);
+        $this->assertEquals($ids[2], $response_data[2]['id']);
+
```
Comment (DarkaOnLine) [inline tests/Feature/PinThreadsTest.php]: Without this, travis is angry (https://travis-ci.org/JeffreyWay/council/jobs/340436609) it looks like it parses json differently :(

File: tests/Feature/PinThreadsTest.php [hunk 5]
```diff
@@ -56,22 +56,16 @@ public function pinned_threads_are_listed_first()
...
-            'data' => [
-                ['id' => $pinned->id],
-                ['id' => $ids[0]],
-                ['id' => $ids[1]]
-            ]
-        ]);
+        $response_data = $this->getJson(route('threads'))->decodeResponseJson()['data'];
+        $this->assertEquals($pinned->id, $response_data[0]['id']);
+        $this->assertEquals($ids[0], $response_data[1]['id']);
+        $this->assertEquals($ids[1], $response_data[2]['id']);
```
Comment (DarkaOnLine) [inline tests/Feature/PinThreadsTest.php]: Without this, travis is angry (https://travis-ci.org/JeffreyWay/council/jobs/340436609) it looks like it parses json differently :(

File: app/Reply.php [hunk 6]
```diff
@@ -43,6 +43,12 @@ protected static function boot()
             $reply->owner->gainReputation('reply_posted');
         });
 
+        static::deleting(function ($reply) {
+            if ($reply->isBest()) {
+                $reply->thread->unsetBestReply();
+            }
+        });
+
```
Comment (DarkaOnLine) [inline app/Reply.php]: Can someone explain this for me :) In the migrations we have foreign key to set `best_reply_id` to `null` on reply delete: https://github.com/JeffreyWay/council/blob/master/database/migrations/2017_03_15_175959_create_threads_table.php#L30-L34

On Laravel 5.5 [this test works](https://github.com/JeffreyWay/council/blob/master/tests/Feature/BestReplyTest.php#L47-L58), but on 5.6 it doesn't :( Why?

**Comment by JeffreyWay** [comment]
Thanks! Will go over this soon.

File: app/Reply.php (hunk 6 above)
Comment (JeffreyWay) [reply app/Reply.php]: You're right, this is really strange! It works with the mysql driver, so seems to be unique to a composer install of 5.6 with the sqlite driver. Weird - will keep researching.

Comment (DarkaOnLine) [reply app/Reply.php]: The problem for this was on Laravel side. RefreshDatabase trait was duplicating transactions.
The fix: https://github.com/laravel/framework/pull/23132/files
More read: https://github.com/laravel/framework/pull/23126

This bus should be fixed on Laravel 5.6.4

**Pull request merged**